Libraries/src/Amazon.Lambda.DynamoDBEvents/Converters/DictionaryLongToStringJsonConverter.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Amazon.Lambda.DynamoDBEvents.Converters
+{
+    public class DictionaryLongToStringJsonConverter : JsonConverter<Dictionary<string, string>>
+    {
+        public override Dictionary<string, string> Read(ref Utf8JsonReader reader, Type type, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.StartObject)
+            {
+                throw new JsonException($"JsonTokenType was of type {reader.TokenType}, only objects are supported.");
+            }
+
+            var dictionary = new Dictionary<string, string>();
+
+            while (reader.Read())
+            {
+                if (reader.TokenType == JsonTokenType.EndObject)
+                {
+                    return dictionary;
+                }
+
+                // Get the key.
+                if (reader.TokenType != JsonTokenType.PropertyName)
+                {
+                    throw new JsonException("JsonTokenType was not PropertyName.");
+                }
+
+                var propertyName = reader.GetString();
+
+                if (string.IsNullOrWhiteSpace(propertyName))
+                {
+                    throw new JsonException("Failed to get property name.");
+                }
+
+                // Get the value.
+                reader.Read();
+                var keyValue = ExtractValue(ref reader, options);
+                dictionary.Add(propertyName, keyValue);
+            }
+
+            return dictionary;
+        }
+
+        public override void Write(Utf8JsonWriter writer, Dictionary<string, string> value, JsonSerializerOptions options)
+        {
+            // Use the built-in serializer, because it can handle dictionaries with string keys.
+            JsonSerializer.Serialize(writer, value, options);
```

Comment:
Wouldn't this write the value as a string but given the context of this converter shouldn't it be written as a long?

Reply:
Yes, it would. I still haven’t received response from Lambda team. My hunch is that the reason there is a note to use `Map<string, string>` for Java is that sometimes the state might contain string values. In such case, we would not be sure on whether to convert it to long or string. That’s the reason I’m handling token type string as well in ExtractValue. Please advise.

Reply:
As discussed, got response from Lambda team that `State` field could have values represented as strings and in some cases long.